content/posts/factorio-gke-spot.md

```diff
@@ -0,0 +1,435 @@
+
+{{< figure src="/images/factorio-gke-spot/factorio.png" width="600px" title="A successful launch is our goal!" >}}
+
+---
+
+## Factorio GKE Server
+
+### What You'll Need (Prerequisites)
+
+*   A [Google Cloud](https://cloud.google.com/?utm_campaign=CDR_0x145aeba1_default_b423920039&utm_medium=external&utm_source=blog)) project with billing enabled.
+*   A Development Environment
+    *   The `gcloud` CLI installed and authenticated.
+    *   The `kubectl` command-line tool installed.
+    *   [Google Cloud Shell Editor](https://ide.cloud.google.com?utm_campaign=CDR_0x145aeba1_default_b423920039&utm_medium=external&utm_source=blog) works great for this, and has the necessary software pre-installed.
+
+---
+
+### Building the Foundation - Your GKE Autopilot Cluster
+
+*   First, ensure the GKE API is enabled on your project. If you haven't used GKE in this project before, you'll need to run this command:
+    ```bash
+    gcloud services enable container.googleapis.com
+    ```
+
+I used GKE autopilot for my cluster so that I don't have to manage nodes myself.  This is particularly handy when migrating to a larger VM, or scaling down to zero when we're not playing for a while.
+
+```bash
+gcloud container clusters create-auto "factorio-autopilot" 
+    --region "us-central1"
+```
+
+(I used `us-central1` as a geographic compromise with friends on the East coast.  Feel free to pick a [different region](https://cloud.google.com/compute/docs/regions-zones?utm_campaign=CDR_0x145aeba1_default_b423920039&utm_medium=external&utm_source=blog) closer to you)
+
+Once your autopilot cluster is up, we'll apply some configuration to it.  This configuration will allow the factorio server to run without needing a persistent server or disk that you have to manage.
+
+
+{{< figure src="/images/factorio-gke-spot/autopilot.png" width="600px" title="A GKE Autopilot Cluster" >}}
+---
+
+### Kubernetes Configuration
+
+#### Config Maps
+
+Since we're running a [generic container](https://hub.docker.com/r/factoriotools/factorio/) of a Factorio server, we need a way to get our game settings on there.  For this, I used Kubernetes Config Maps based on files on my computer (in this case, Cloud Shell).
```

Comment:
```suggestion
Since we're running a [generic container](https://hub.docker.com/r/factoriotools/factorio/) of a Factorio server, we need a way to get our game settings on there.  For this, I used Kubernetes ConfigMaps based on files on my computer (in this case, Cloud Shell).
```